From 0d3a8f0099f0d4b89714c3dc5c6a8daa1ff97be1 Mon Sep 17 00:00:00 2001
From: cchheennhhaaoo <hao3.chen@intel.com>
Date: Tue, 21 Jun 2022 18:16:02 +0000
Subject: [PATCH 17/29] Enhance negative operator for SYCL half conversion
 (#79850)

Enhance negative operator for SYCL half conversion with sycl::bit_cast implementation.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/79850
Approved by: https://github.com/ngimel
---
 c10/util/Half-inl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/c10/util/Half-inl.h b/c10/util/Half-inl.h
index f634afd7df..3801080fb5 100644
--- a/c10/util/Half-inl.h
+++ b/c10/util/Half-inl.h
@@ -93,7 +93,7 @@ inline C10_HOST_DEVICE Half operator-(const Half& a) {
     defined(__HIP_DEVICE_COMPILE__)
   return __hneg(a);
 #elif defined(__SYCL_DEVICE_ONLY__)
-  return -static_cast<sycl::half>(a);
+  return -sycl::bit_cast<sycl::half>(a);
 #else
   return -static_cast<float>(a);
 #endif
-- 
2.25.1

